### PR TITLE
feat: add support for tcp endpoints in haproxy plugin

### LINKED ISSUE #12672: Support stats socket with tcp address in haproxy input plugin
### Use Case

It is possible to configure haproxy stats endpoint to be a socket file (implemented) or tcp address (not implemented, docs https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#3.1-stats%20socket).

Having it as tcp address is a supported option and might be more comfortable to use than socket file (for example when running haproxy in container)

### Expected behavior

When using the plugin with the following configuration:
```
[[inputs.haproxy]]
  servers = ["tcp://localhost:1936"]
```
it will be able to collect statistics

### Actual behavior

It tries to connect to unix socket with path 1936

### Additional info

Could be implemented by updating server parser and removing hardcoded network argument `unix` here https://github.com/influxdata/telegraf/blob/master/plugins/inputs/haproxy/haproxy.go#L90

### PR DESCRIPTION
**Pull request by jsnjack**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #12672 

- Added support for tcp endpoints in haproxy input plugin. For example, if haproxy is configured to expose stats endpoint with `stats socket ipv4@127.0.0.1:9999 level admin`, the plugin will be able to communicate with it with `servers = ["tcp://127.0.0.1:9999"]` configuration

### CODE REVIEW

**Comment by powersj** [comment]
> Could not find a usable config.yml, you may have revoked the CircleCI OAuth app.

This error from CircleCI shows up when you have an account with CircleCI and either you are logged out if you use a username/password or if you have a GitHub OAuth login, need to disconnect and reconnect your account.

File: plugins/inputs/haproxy/sample.conf [hunk 1]
```diff
@@ -10,10 +10,10 @@
...
 
   ## If no servers are specified, then default to 127.0.0.1:1936/haproxy?stats
-  servers = ["http://myhaproxy.com:1936/haproxy?stats"]
+  servers = ["http://myhaproxy.com:1936/haproxy?stats", "tcp://127.0.0.1:1936"]
 
   ## You can also use local socket with standard wildcard globbing.
-  ## Server address not starting with 'http' will be treated as a possible
+  ## Server address not starting with 'http://', 'https://', 'tcp://' will be treated as a possible
   ## socket, so both examples below are valid.
   # servers = ["socket:/run/haproxy/admin.sock", "/run/haproxy/*.sock"]
```
Comment (powersj) [inline plugins/inputs/haproxy/sample.conf:18]: The sample config values should indicate the default value. Most users tend to stop reading examples once they have come across one, so the second paragraph showing the socket examples, is often ignored or missed. Let's combine these two sections into something like:

```
  ## URLs to gather stats from
  ## For HTTP and HTTPS URLs, the URL URL must specified the complete path to
  ## the endpoint containing stats:
  ##   - http://myhaproxy.com:1936/haproxy?stats
  ##   - https://myhaproxy.com:8000/stats
  ## The plugin can also connect to a socket:
  ##   - socket:/run/haproxy/admin.sock
  ##   - tcp://127.0.0.1:1936
  ##   - /run/haproxy/*.sock
  # servers = ["http://127.0.0.1:1936/haproxy?stats"]
```

**Review by powersj — COMMENTED** [review]
Thanks for taking the time to put up a PR. I have one small nit and a doc change that I'd like to see. Feel free to wordsmith or update that as you see fit.

Thanks!

File: plugins/inputs/haproxy/sample.conf (hunk 1 above)
Comment (jsnjack) [reply plugins/inputs/haproxy/sample.conf:22]: Thanks! Good points!

File: plugins/inputs/haproxy/sample.conf [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+  ##   - socket:/run/haproxy/admin.sock
+  ##   - /run/haproxy/*.sock
+  ##   - tcp://127.0.0.1:1936
+  ##
+  ## Server addresses not starting with 'http://', 'https://', 'tcp://' will be
+  ## treated as possible sockets.
+  ##
+  ## When specifying local socket, glob patterns are supported.
+  ##
+  ## If no servers are specified, then defaults to 127.0.0.1:1936/haproxy?stats
```
Comment (powersj) [inline plugins/inputs/haproxy/sample.conf:16]:
```suggestion
  ## Server addresses not starting with 'http://', 'https://', 'tcp://' will be
  ## treated as possible sockets. When specifying local socket, glob patterns
  ## are supported.
```

File: plugins/inputs/haproxy/sample.conf [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
   servers = ["http://myhaproxy.com:1936/haproxy?stats"]
 
   ## You can also use local socket with standard wildcard globbing.
-  ## Server address not starting with 'http' will be treated as a possible
+  ## Server address not starting with 'http://', 'https://', 'tcp://' will be treated as a possible
   ## socket, so both examples below are valid.
   # servers = ["socket:/run/haproxy/admin.sock", "/run/haproxy/*.sock"]
```
Comment (powersj) [inline plugins/inputs/haproxy/sample.conf:22]: Can you get rid of this second section and run `make docs` please. Thanks!

Comment (jsnjack) [reply plugins/inputs/haproxy/sample.conf:19]: This was done in a previous commit

**Review by powersj — APPROVED** [review]
Thanks for putting this up!

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for your contribution @jsnjack!

**Pull request merged**